### ISSUE TITLE
Can't find keyboard layout on Windows under Boot Camp with only Apple keyboard layout

### ISSUE THREAD

**Issue by jwwalker** [symless/synergy-core#5176]
4 users on the forum have reported this recently: http://synergy-project.org/forum/viewtopic.php?f=16&t=1079

**Comment by timmerk** [comment]
Confirmed - this started happening today for me. Restarting both Win client and Mac server doesn't fix. Mouse still works between screens, but Mac keys are not sent to Windows.

**Comment by jwwalker** [comment]
Here's a snippet from the log.  Looks like the server "thinks" it is sending key events.

[2016-01-08T09:45:22] DEBUG1: event: Key event kind: 10, keycode=126
[2016-01-08T09:45:22] DEBUG1: new mask: 0x0000
[2016-01-08T09:45:22] DEBUG1: onKeyDown id=61266 mask=0x0000 button=0x007f
[2016-01-08T09:45:22] DEBUG1: send key down to "vistameister" id=61266, mask=0x0000, button=0x007f
[2016-01-08T09:45:22] DEBUG2: writef(DKDN%2i%2i%2i)
[2016-01-08T09:45:22] DEBUG2: wrote 10 bytes
[2016-01-08T09:45:22] DEBUG2: msg from "vistameister": CNOP
[2016-01-08T09:45:22] DEBUG2: no-op from

**Comment by jwwalker** [comment]
And here's a bit of the log from the client.

[2016-01-08T10:06:26] DEBUG2: msg from server: DKDN
[2016-01-08T10:06:26] DEBUG2: readf(%2i%2i%2i)
[2016-01-08T10:06:26] DEBUG2: readf: read 2 byte integer: 61266 (0xef52)
[2016-01-08T10:06:26] DEBUG2: readf: read 2 byte integer: 0 (0x0)
[2016-01-08T10:06:26] DEBUG2: readf: read 2 byte integer: 127 (0x7f)
[2016-01-08T10:06:26] DEBUG1: recv key down id=0x0000ef52, mask=0x0000, button=0x007f
[2016-01-08T10:06:26] DEBUG1: can't find keyboard layout 04090409
[2016-01-08T10:06:26] DEBUG1: mapKey ef52 (61266) with mask 0000, start state: 0000
[2016-01-08T10:06:26] DEBUG1: key ef52 is not on keyboard

**Comment by jwwalker** [comment]
Going the other way, Windows server to Mac client, sort of works.  The shift key does not work, nor does the option key, but the caps lock key does.

**Comment by XinyuHou** [comment]
Duplicate to #626

**Comment by jwwalker** [comment]
@XinyuHou, I am not convinced this is the same bug as #626.  In that bug, keathmilligan said that no keyboard events were being sent to the client, and it seemed to be fixed in 1.7.1.  In this bug, keyboard events are being received by the client but not handled correctly, and the bug is present in 1.7.5.

**Comment by XinyuHou** [comment]
@jwwalker 
What keyboard layout do you use on server and client?

**Comment by jwwalker** [comment]
@XinyuHou 
On the Mac server, I use the "U.S. Extended" layout.

The client is a Mac running Windows Vista under Boot Camp. In the Text Services and Input Languages inspector, it said that the input language was "English (United States) - United States (Apple)", and that was the only choice available.  So while looking for the answer to your question, I clicked the "Add..." button, and in the resulting "Add Input Language" window, under "English (United States)", I checked the "US" layout.  And now Synergy processes keystrokes correctly!  I don't have to actually select the "US" keyboard layout, it just needs to be available.  Thank you for getting back to me and asking the right question.

**Comment by XinyuHou** [comment]
@jwwalker 
Would you mind share your solution on the user wiki? Here is the link.
https://github.com/symless/synergy/wiki/User-Guide

**Comment by jwwalker** [comment]
@XinyuHou 
OK, I added a paragraph to the User FAQ under the existing heading "Why doesn't my keyboard work on the client?"

**Comment by XinyuHou** [comment]
@jwwalker 
Thanks

**Comment by timmerk** [comment]
This issue isn't closed, as I'm sure you know. It's not like all of us turn on and off the U.S. input source in between when it works and doesn't work. I've never touched the settings, and mine is set to U.S. I get this issue about once a week.

**Comment by jwwalker** [comment]
@timmerk 
When you get this problem, if you turn the client logging up to Debug1 or more, do you get the "can't find keyboard layout" message that I saw?

**Comment by timmerk** [comment]
No, I don't get that, but I have the same issue where keys are received but never sent.

**Comment by jwwalker** [comment]
@timmerk 
I assume you mean keys are sent but never received.  But that's not the right description of the error I experienced.  In my case the key events were sent **and** received but then dropped on the floor.  So maybe you have a different problem than I did, and maybe you should post what you see in the logs.

**Comment by the-wes** [comment]
The root cause may still be a duplicate of other bugs, but since this is a distinct presentation, we'll leave it open for now.

**Comment by brendensoares** [comment]
I was using synergy fine for some time (Mac 12.2 server/synergy 1.14.2 on Windows 10 client/synergy 1.4.13-rc) and then this morning the keyboard stopped working. Not sure what changed.

Restarting client and server doesn't help. On the client side I keep getting notifications when my mouse enters the client from the server:

```
The keyboard may stop working.
---
"Secure input" enabled by loginwindow on the server. To fix the keyboard, loginwindow must be closed.

```
Why is this ticket stale? Why was a solution not found? This seems to [affect Windows](https://github.com/symless/synergy-core/issues/5176#issuecomment-179391630) and Mac.

Hmm...sleep mode?
---
While I was writing this reply, I had to step away. When I came back my Mac was asleep. I woke it up. Now the keys are working on the Windows client.

Note, I have a USB keyboard hooked up through a USB KVM (backup in case synergy isn't working) which is connected through a USB hub. The KVM has changed how my keyboard works when the Macbook sleeps (LED lights go off). Not sure why.

TL;DR: this is still an issue and may be caused by power hibernation/sleep mode
